REST/PowerShell/Variables/ChangeVariableValueinVariableSet.ps1

```diff
@@ -0,0 +1,92 @@
+#Allows HTTPS connections
+[Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12
+
+$VarSetName = "YourVariableSetName"           #The exact name of the Variable Set you are trying to change a variable within. Not needed if you know ID
+$VarSetID = ""                                #Leave blank if you dont know your variable set ID
+
+$VariableName = "YourVariableName"            #The name of the variable you are trying to change the value of
+$VariableValue = "YourValue"                  #New value of Variable
+$VariableEnvName = "YourEnvironment"          #Scope of a variable to a environment, leave blank to scope to all
+$OverrideMultipleEnv = $False                 #True will replace Value for environments will multiple scopes i.e. if a environment is part of 2 scopes it will replace both 
+                                              #False will cause an error if the environment is shared with more than value or if multiple scopes have the environment
+
+$OctopusSite = "https://your.site.com"        #Your Octopus site name
+$octoapikey = "API-................"          #Your Octopus API Key
+
+
+#If the Library ID was not provided it will try and find one instance of it using the Library Name
+if (!$VarSetID) {
+    try {
+        $VarSets = Invoke-RestMethod "$OctopusSite/api/libraryvariablesets" -Method Get -Headers @{"X-Octopus-ApiKey" = $octoapikey} -ContentType "application/json"
+        $VarSetID = ($VarSets.Items | Where-Object { $_.Name -ieq $VarSetName }).ID
+        if (($VarSetID).count -lt 1) {
+            Write-Error "Did not find any entries matching that Library name"
+            return -1
+        }
+        elseif (($VarSetID).count -gt 1) {
+            Write-Error "Found more than one entry matching that Library name"
+            return -1
+        }
+    }
+    catch {
+        Write-Error $_
+        Return -1
+    }
+}
+
+#Pulls down all variables from Library ID. Note sensitive values are not exposed
+try {
+    $Variables = Invoke-RestMethod "$OctopusSite/api/variables/variableset-$VarSetID" -Method Get -Headers @{"X-Octopus-ApiKey" = $octoapikey} -ContentType "application/json"
+}
+catch {
+    Write-Error $_
+    Return -1
+}
+
+#Finds the Environment ID if Environment Name is provided
+if ($VariableEnvName) {
+    $VariableEnvID = ($Variables.ScopeValues.Environments | Where-Object { $_.Name -ieq $VariableEnvName}).ID
```

Comment:
You need to initialize `$VariableEnvID` to `$null` before this line, otherwise if you run the script with an environment name specified and then again for a different variable without an environment name the script will still think it's looking for an environment down on line 61 and throw an error if the second variable does not have a scope to that environment.